AlttpRandomizer/Rom/RomLocationsCasual.cs

```diff
@@ -103,11 +103,12 @@ public void ResetLocations()
                     CanAccess =
                         have =>
                         CanEnterSwampPalace(have) 
-                        && have.Contains(InventoryItemType.Hammer) 
-                        && (have.Contains(InventoryItemType.Hookshot) 
-                            || (!LocationHasItem("[dungeon-D2-B2] Swamp Palace - flooded room [left chest]",InventoryItemType.BigKey) 
-                                && !LocationHasItem("[dungeon-D2-B2] Swamp Palace - flooded room [right chest]",InventoryItemType.BigKey)   
-                                && !LocationHasItem("[dungeon-D2-B2] Swamp Palace - hidden waterfall door room",InventoryItemType.BigKey))),
+                        && have.Contains(InventoryItemType.Hammer)
+                        && ((LocationHasItem("[dungeon-D2-B1] Swamp Palace - big key room", InventoryItemType.BigKey)
+                                || LocationHasItem("[dungeon-D2-B1] Swamp Palace - map room", InventoryItemType.BigKey)
+                                || LocationHasItem("[dungeon-D2-B1] Swamp Palace - push 4 blocks room", InventoryItemType.BigKey)
+                                || LocationHasItem("[dungeon-D2-B1] Swamp Palace - south of hookshot room", InventoryItemType.BigKey))
+                            || (have.Contains(InventoryItemType.Hookshot) && !LocationHasItem("[dungeon-D2-B1] Swamp Palace - big chest",InventoryItemType.Hookshot)))

```

Comment:
This is on the Big Chest logic

First it checks to see if the Big Key has been placed in the chests you can access without the hoookshot.

If it's not in any of those chests, it checks to see if you have access to the hookshot. If you don't it returns False telling the Randomizer you can't access the big chest.

If you do have access to the hookshot, it also checks to make sure it's not in the big chest itself.  If it's not, again you can access the last three chests and get the big key if it were in there.

If the big key hasn't been placed at all it already shouldn't think you can access it anyway.

It's not perfect but it's the best method i've been able to come up with that didn't generate RandomizationExceptions when ingenerated 100 seeds. I want to find a better way, and plan to continue looking at it but at least for now I'm happy with the generation for my own use (wouldn't dare use it to race against others for obvious reasons)

If anything stands out as weird or if you have ideas for improvements I'd love to hear it. Been a great learning process, but tons more to go.